src/fondant/dataset/dataset.py

```diff
@@ -440,30 +440,62 @@ def get_nested_dict_hash(input_dict):
         return get_nested_dict_hash(component_op_uid_dict)
 
 
-class Pipeline:
-    """Class representing a Fondant Pipeline."""
+class Workspace:
+    """Workspace holding environment information for a Fondants execution environment."""
 
     def __init__(
         self,
         name: str,
-        *,
         base_path: str,
         description: t.Optional[str] = None,
     ):
-        """
-        Args:
-            name: The name of the pipeline.
-            base_path: The base path for the pipeline to use to store artifacts and data. This
-                can be a local path or a remote path on one of the supported cloud storage
-                services. The path should already exist.
-            description: Optional description of the pipeline.
-        """
-        self.base_path = base_path
-        self.name = self._validate_pipeline_name(name)
+        self.name = self._validate_workspace_name(name)
         self.description = description
+        self.base_path = base_path
         self.package_path = f"{name}.tgz"
+
+    @staticmethod
+    def _validate_workspace_name(name: str) -> str:
+        pattern = r"^[a-z0-9][a-z0-9_-]*$"
+        if not re.match(pattern, name):
+            msg = f"The workspace name violates the pattern {pattern}"
+            raise InvalidWorkspaceDefinition(msg)
+        return name
+
+    def get_run_id(self) -> str:
```

Comment:
Not sure if the caching is currently working. 
The run id is used to determine changes. The workspace should hold run ids in the future as well. I could think about a combination of runner type and working directory.

Reply:
I would not put more effort into it for this PR. we can test and fix separately